components/server/src/ome/services/sessions/SessionManagerImpl.java

```diff
@@ -519,9 +519,36 @@ protected SessionContext createSessionContext(List<?> list, SessionContext previ
 
     public Session find(String uuid) {
         SessionContext sessionContext = cache.getSessionContext(uuid);
+        checkIfShare(sessionContext);
         return (sessionContext == null) ? null : sessionContext.getSession();
     }
 
+    private void checkIfShare(SessionContext sessionContext) {
+        if (sessionContext.getSession() instanceof Share) {
+            final Long id = sessionContext.getSession().getId();
+            final String uuid = sessionContext.getSession().getUuid();
+            final String prefix = String.format("Share:%s (%s)", id, uuid);
+
+            List<Object[]> rv = executeProjection(
+                    "select s.active, s.timeToLive, s.started from Share s where s.id = :id",
+                    new Parameters().addId(sessionContext.getSession().getId()));
+
+            if (rv.size() != 1) {
+                throw new RuntimeException(prefix + " could not be found!");
+            }
+
+            Object[] items = rv.get(0);
+            Boolean active = (Boolean) items[0];
+            Long timeToLive = (Long) items[1];
+            Timestamp started = (Timestamp) items[2];
+
+            if (Boolean.FALSE.equals(active)) {
+               throw new SecurityViolation(prefix + " is inactive");
+            } else if (started.getTime() + timeToLive > System.currentTimeMillis()) {
```

Comment:
There's never no expiry but it is set to `Long.MAX_VALUE` which is causing an overflow. I'll push a fix for that in just a second.